os_migrate/plugins/module_utils/network.py

```diff
@@ -156,3 +156,140 @@ def network_refs_from_ser(conn, ser_net):
+        'project_id',
+        'created_at',
+        'updated_at',
+    ])
+
+    set_ser_params_same_name(params, sdk_sec_rule, [
+        'description',
+        'direction',
+        'port_range_max',
+        'port_range_min',
+        'protocol',
+        'remote_ip_prefix',
+        'revision_number',
+    ])
+
+    return resource
+
+
+def serialize_security_group(sdk_sec):
+    """Serialize OpenStack SDK security group `sdk_sec`
+    into OS-Migrate format.
+
+    Returns: Dict - OS-Migrate structure for Security group.
+    """
+    expected_type = openstack.network.v2.security_group.SecurityGroup
+    if type(sdk_sec) != expected_type:
+        raise exc.UnexpectedResourceType(expected_type, type(sdk_sec))
+
+    resource = {}
+    params = {}
+    info = {}
+    sdk_params = {}
+
+    resource[const.RES_PARAMS] = params
+    resource[const.RES_INFO] = info
+    resource[const.RES_TYPE] = const.RES_TYPE_SECURITYGROUP
+    params['tags'] = sorted(sdk_sec['tags'])
+
+    # Decide which attrs are param and which are info
+    set_ser_params_same_name(info, sdk_sec, [
+        'id',
+        'tenant_id',
+        'project_id',
+        'created_at',
+        'updated_at',
+    ])
+
+    set_ser_params_same_name(params, sdk_sec, [
+        'name',
+        'description',
+        'revision_number',
+        'project_id',
+        'location',
+    ])
+
+    return resource
+
+
+def security_group_sdk_params(ser_sec):
+    """Create OpenStack SDK parameters dict for creation or update of the
+    OS-Migrate Security groups `ser_sec`.
+
+    Returns: Parameters to be fed into `openstack.network.v2.security_group.SecurityGroup`
+    """
+    res_type = ser_sec.get(const.RES_TYPE, None)
+    if res_type != const.RES_TYPE_SECURITYGROUP:
+        raise exc.UnexpectedResourceType(const.RES_TYPE_SECURITYGROUP, res_type)
+
+    ser_params = ser_sec[const.RES_PARAMS]
+    sdk_params = {}
+
+    set_sdk_params_same_name(ser_params, sdk_params, [
+        'description',
+        'direction',
+        'port_range_max',
+        'port_range_min',
+        'protocol',
+        'remote_ip_prefix',
+        'revision_number',
+    ])
+
+    return sdk_params
+
+
+def security_group_rule_sdk_params(ser_sec_rule):
+    """Create OpenStack SDK parameters dict for creation or update of the
+    OS-Migrate Security group rules `ser_sec_rule`.
+
+    Returns: Parameters to be fed into `openstack.network.v2.security_group_rule.SecurityGroupRule`
+    """
+    res_type = ser_sec_rule.get(const.RES_TYPE, None)
+    if res_type != const.RES_TYPE_SECURITYGROUPRULE:
+        raise exc.UnexpectedResourceType(const.RES_TYPE_SECURITYGROUPRULE, res_type)
+
+    ser_params = ser_sec_rule[const.RES_PARAMS]
+    sdk_params = {}
+
+    set_sdk_params_same_name(ser_params, sdk_params, [
+        'description',
+        'direction',
+        'port_range_max',
+        'port_range_min',
+        'protocol',
+        'remote_ip_prefix',
+        'revision_number',
```

Comment:
I'd add `# FIXME: missing remote_group_id and security_group_id`. Without security_group_id, imports will not work. But it's probably fine to merge so that we have a baseline merged.